c7n/resources/appmesh.py

```diff
@@ -0,0 +1,260 @@
+"""
+AppMesh Communications
+"""
+from c7n import query
+from c7n.exceptions import PolicyExecutionError
+from c7n.manager import resources
+from c7n.tags import universal_augment
+from c7n.query import (
+    ChildResourceManager,
+    QueryResourceManager,
+    TypeInfo,
+    DescribeSource,
+    ChildDescribeSource,
+    ConfigSource,
+)
+from c7n.resources.aws import Arn
+from c7n.utils import local_session
+
+
+class DescribeMesh(DescribeSource):
+    # override default describe augment to get tags
+    def augment(self, resources):
+        return universal_augment(self.manager, resources)

```

Comment:
custodian defaults to fetching tags for resources, this describe source customization ensure that for this resource type.